### PR TITLE
Migrate PerformanceTest plugin to Kotlin

### PR DESCRIPTION
**Pull request by bamboo**

### CODE REVIEW

File: buildSrc/src/main/kotlin/org/gradle/plugins/PerformanceTestPlugin.kt [hunk 1]
```diff
@@ -132,16 +132,15 @@ class PerformanceTestPlugin : Plugin<Project> {
...
+            coordinatorBuildId = stringPropertyOrNull(coordinatorBuildIdProperty)
+            branchName = stringPropertyOrNull(branchnameProperty)
             teamCityUrl = teamCityUrlValue
-            teamCityUsername = propertyOrNull(teamCityUserNameProperty)
-            teamCityPassword = propertyOrNull(teamCityPasswordProperty)
+            teamCityUsername = stringPropertyOrNull(teamCityUserNameProperty)
+            teamCityPassword = stringPropertyOrNull(teamCityPasswordProperty)
             afterEvaluate {
-                if (branchName != null && !branchName.isEmpty()) {
+                branchName?.takeIf { it.isNotEmpty() }?.let { branchName ->
```
Comment (hansd) [inline buildSrc/src/main/kotlin/org/gradle/plugins/PerformanceTestPlugin.kt]: Doesn't this cause a NPE if branchName is null?

Comment (bamboo) [reply buildSrc/src/main/kotlin/org/gradle/plugins/PerformanceTestPlugin.kt]: It doesn't thanks to the safe-navigation operator `?.` combined with the `takeIf` and `let` primitives:

    e?.takeIf(pred) => if (e != null && pred(e)) e else null
    e?.let(f) => if (e != null) f(e) else null

Comment (hansd) [reply buildSrc/src/main/kotlin/org/gradle/plugins/PerformanceTestPlugin.kt]: Very nice. Thanks.

**Comment by bamboo** [comment]
CI build: https://builds.gradle.org/viewQueued.html?itemId=11140548&tab=queuedBuildOverviewTab

**Pull request merged**